lib/cache/cache.go

```diff
@@ -2491,6 +2516,32 @@ func (c *Cache) GetAccessList(ctx context.Context, name string) (*accesslist.Acc
 	return rg.reader.GetAccessList(ctx, name)
 }
 
+// GetUserLoginStates returns the all user login state resources.
+func (c *Cache) GetUserLoginStates(ctx context.Context) ([]*userloginstate.UserLoginState, error) {
+	ctx, span := c.Tracer.Start(ctx, "cache/GetUserLoginStates")
+	defer span.End()
+
+	rg, err := readCollectionCache(c, c.collections.userLoginStates)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	defer rg.Release()
+	return rg.reader.GetUserLoginStates(ctx)
+}
+
+// GetUserLoginState returns the specified user login state resource.
+func (c *Cache) GetUserLoginState(ctx context.Context, name string) (*userloginstate.UserLoginState, error) {
+	ctx, span := c.Tracer.Start(ctx, "cache/GetUserLoginState")
+	defer span.End()
+
+	rg, err := readCollectionCache(c, c.collections.userLoginStates)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	defer rg.Release()
+	return rg.reader.GetUserLoginState(ctx, name)
+}
```

Comment:
Take a look at the implementation of `Cache.GetUser`.  You'll notice that if it gets a `NotFound` error from a cache read, it will explicitly try reading from the upstream collection.  We do this because sometimes there is a slight delay after login where the updated user resource hasn't been propagated to the cache (this can be hard to replicate locally, but it definitely happens in real clusters that are under medium to high load).

Now that login state is going to replace a lot of those reads, we should do the same thing here.